### ISSUE TITLE
Empty Mask Issue

### ISSUE THREAD

**Issue by MarcelMoczarski** [lquesada/ComfyUI-Inpaint-CropAndStitch#168]
Hey, first of all great node I use it a lot for inpainting.

However, I’ve run into an issue with the improved version. When I use a preview bridge node (Impact Pack) without a mask, it throws an error. The preview bridge outputs the mask as a 2D tensor (for example, (64, 64) and not (1, 64, 64)) when no mask is used.

Your inpaint crop node doesn’t check if the image and mask dimensions match. Because of that, in this case it treats the mask as a batch of 64 images (since it expects something like (1, h, w)), which makes it iterate over them and takes a very long time.

This doesn’t happen when using a normal load image node, because there the empty mask has the correct shape ((1, h, w) instead of (h, w)).

I’m not sure if this behavior is intended, but for now I added a simple fix that raises an error if the image and mask dimensions don’t match. Here is Claudes Fix:

Replaced (original code, lines ~1303-1314):
With (new code):

```python
# Validate mask tensor dimensions: must be 3D (B, H, W) and spatial dims must match image
if mask is not None:
    if mask.ndimension() == 2:
        raise ValueError(
            f"Inpaint Crop: Mask input has 2 dimensions {tuple(mask.shape)}, expected 3 dimensions (batch, height, width). "
            f"The mask spatial size ({mask.shape[0]}x{mask.shape[1]}) does not match the image size ({image.shape[1]}x{image.shape[2]}). "
            f"This typically happens when no mask has been painted yet. Please draw a mask before connecting it."
        )
    if mask.ndimension() != 3:
        raise ValueError(
            f"Inpaint Crop: Mask input has {mask.ndimension()} dimensions {tuple(mask.shape)}, expected 3 dimensions (batch, height, width)."
        )
    if mask.shape[1] != image.shape[1] or mask.shape[2] != image.shape[2]:
        if torch.count_nonzero(mask) == 0:
            mask = torch.zeros((mask.shape[0], image.shape[1], image.shape[2]), device=image.device, dtype=image.dtype)
        else:
            raise ValueError(
                f"Inpaint Crop: Mask spatial dimensions ({mask.shape[1]}x{mask.shape[2]}) do not match image dimensions ({image.shape[1]}x{image.shape[2]})."
            )

if optional_context_mask is not None:
    if optional_context_mask.ndimension() == 2:
        raise ValueError(
            f"Inpaint Crop: Optional context mask has 2 dimensions {tuple(optional_context_mask.shape)}, expected 3 dimensions (batch, height, width). "
            f"Please draw a mask before connecting it."
        )
    if optional_context_mask.ndimension() != 3:
        raise ValueError(
            f"Inpaint Crop: Optional context mask has {optional_context_mask.ndimension()} dimensions {tuple(optional_context_mask.shape)}, expected 3 dimensions (batch, height, width)."
        )
    if optional_context_mask.shape[1] != image.shape[1] or optional_context_mask.shape[2] != image.shape[2]:
        if torch.count_nonzero(optional_context_mask) == 0:
            optional_context_mask = torch.zeros((optional_context_mask.shape[0], image.shape[1], image.shape[2]), device=image.device, dtype=image.dtype)
        else:
            raise ValueError(
                f"Inpaint Crop: Optional context mask spatial dimensions ({optional_context_mask.shape[1]}x{optional_context_mask.shape[2]}) do not match image dimensions ({image.shape[1]}x{image.shape[2]})."
            )
```

**Comment by lquesada** [comment]
thanks, i'll look into this!

**Comment by lquesada** [comment]
I added a fix for this input, please update :)
